pkg/cloudprovider/providers/openstack/openstack_routes.go

```diff
@@ -195,7 +200,7 @@ func (r *Routes) CreateRoute(clusterName string, nameHint string, route *cloudpr
 		newPairs := append(port.AllowedAddressPairs, neutronports.AddressPair{
 			IPAddress: route.DestinationCIDR,
 		})
-		unwind, err := updateAllowedAddressPairs(r.network, &port, newPairs)
+		unwind, err := updateAllowedAddressPairs(r.network, port, newPairs)
```

Comment:
It is need to check the AllowedAddressPairs of port before calling updateAllowedAddressPairs.
And updateAllowedAddressPairs will call port.AllowedAddressPairs too.